geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/sql/GeoMesaSparkSql.scala

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright (c) 2013-2015 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0 which
+ * accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ */
+
+package org.locationtech.geomesa.compute.spark.sql
+
+import java.sql.Timestamp
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.{Date, List => jList, Map => jMap, UUID}
+
+import com.typesafe.scalalogging.slf4j.Logging
+import com.vividsolutions.jts.geom.Geometry
+import org.apache.hadoop.conf.Configuration
+import org.apache.metamodel.{query, DataContext}
+import org.apache.metamodel.query.FilterClause
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{Row, _}
+import org.apache.spark.{SparkConf, SparkContext}
+import org.geotools.data.{DataUtilities, DataStoreFinder, Query}
+import org.geotools.factory.CommonFactoryFinder
+import org.geotools.filter.text.ecql.ECQL
+import org.geotools.filter.visitor.DuplicatingFilterVisitor
+import org.locationtech.geomesa.compute.spark.GeoMesaSpark
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes._
+import org.locationtech.geomesa.utils.text.WKTUtils
+import org.opengis.feature.`type`.AttributeDescriptor
+import org.opengis.feature.simple.SimpleFeatureType
+import org.opengis.filter.Filter
+import org.opengis.filter.expression.PropertyName
+
+import scala.collection.JavaConversions._
+
+/**
+ * Class to manage running sql queries against geomesa using spark.
+ *
+ * There can only be a single spark context running in a given jvm, so it has to be managed using the
+ * start/stop/register methods.
+ */
+object GeoMesaSparkSql extends Logging {
+
+  private val ff = CommonFactoryFinder.getFilterFactory2
+
+  // state to keep track of our sfts and data store connection parameters
+  private val sftsByName = scala.collection.mutable.Map.empty[String, SimpleFeatureType]
+  private val paramsBySft = scala.collection.mutable.Map.empty[SimpleFeatureType, Map[String, String]]
+
+  // singleton spark context
+  private var sc: SparkContext = null
```

Comment:
question...if I am running a spark shell where I already have my SparkContext can I still make use of GeoMesaSparkSql? Perhaps having an apply() method that takes the SC in would work?